plugins/modules/identity/ipa/ipa_config.py

```diff
@@ -22,25 +29,156 @@
     description: Default e-mail domain for new users.
     aliases: ["emaildomain"]
     type: str
+  ipadefaultprimarygroup:
+    description: Default group for new users.
+    aliases: ["primarygroup"]
+    type: str
+    version_added: '2.5.0'
+  ipagroupsearchfields:
+    description: A comma-separated list of fields to search in when searching for groups.
+    aliases: ["groupsearchfields"]
+    type: str
```

Comment:
Wouldn't it make sense to make this a list of strings as well, and convert it to the correct format in the module?

Reply:
While it would, IPA doesn't seem to support it.  I tried that way first, but according to IPA, this field is not 'multivalued' - just a str field that has, well multiple values.  I thought the ipa.py may just make this happen behind the scenes turning it into a comma delimited list, but it doesn't.  IPA also returns it a single string and not an set of values, so it is hard in both directions.  Since the IPA API does not list it as multi-select, I just decided to play along with it and make it a str field to match the API.

Reply:
The module's API must not be a 100% match to the service's API. The module can do some translation work, like converting a list of strings to a comma-separated string.

Reply:
Resolved in d031577

Reply:
Thanks! I think this definitely improves the user interface.